src/ExactDiagonalization/init_and_solvers.jl

```diff
@@ -0,0 +1,149 @@
+"""
+    init(p::ExactDiagonalizationProblem, [algorithm]; kwargs...)
+
+Initialize a solver for an [`ExactDiagonalizationProblem`](@ref) `p` with an optional
+`algorithm`. Returns a solver instance that can be solved with [`solve`](@ref).
+
+For a description of the keyword arguments, see the documentation for
+[`ExactDiagonalizationProblem`](@ref).
+"""
+function CommonSolve.init( # no algorithm specified as positional argument
+    p::ExactDiagonalizationProblem;
+    kwargs...
+)
+    kw_nt = (; p.kw_nt..., kwargs...) # remove duplicates
+    if isdefined(kw_nt, :algorithm)
+        algorithm = kw_nt.algorithm
+        kw_nt = delete(kw_nt, (:algorithm,))
+    else
+        algorithm = LinearAlgebraSolver()
+    end
```

Comment:
```suggestion
    algorithm = get(kw_nt, :algorithm, LinearAlgebraSolver())
    kw_nt = delete(kw_nt, (:algorithm,))
```
I'd prefer it written like this, but if you find the original more readable, that's fine too,